grpc-circuitbreaker-utils/src/main/java/org/hypertrace/circuitbreaker/grpcutils/resilience/ResilienceCircuitBreakerProvider.java

```diff
@@ -17,51 +19,42 @@ class ResilienceCircuitBreakerProvider {
   private static final String SHARED_KEY = "SHARED_KEY";
   private final CircuitBreakerRegistry circuitBreakerRegistry;
   private final Map<String, CircuitBreakerConfig> circuitBreakerConfigMap;
-  private final Map<String, CircuitBreaker> circuitBreakerCache = new ConcurrentHashMap<>();
   private final List<String> disabledKeys;
   private final boolean defaultEnabled;
 
+  // LoadingCache to manage CircuitBreaker instances with automatic loading and eviction
+  private final LoadingCache<String, Optional<CircuitBreaker>> circuitBreakerCache =
+      CacheBuilder.newBuilder()
+          .expireAfterAccess(60, TimeUnit.MINUTES) // Auto-evict after 60 minutes
+          .maximumSize(10000) // Limit max cache size
+          .build(
+              new CacheLoader<>() {
+                @Override
+                public Optional<CircuitBreaker> load(String key) {
+                  return buildNewCircuitBreaker(key);
+                }
+              });
```

Comment:
nit
```suggestion
          .build(CacheLoader.from(this::buildNewCircuitBreaker));
```

Reply:
Will fix in immediate PR, merging this

Reply:
https://github.com/hypertrace/java-grpc-utils/pull/69/files Please review